### PR TITLE
Update dependencies and minor fixes

### PR DESCRIPTION
**Pull request by rchildre3**
* Use dependabot to keep dependencies up-to-date, fairly lax monthly schedule, can update to be more frequent, if necessary.
* Example testcase suggests grammar which will never find coverage (xml) generator. move to the regex mutator to at least find something.
* `serde_derive` crate is now a feature of the main serde crate.
  * Add necessary use statements
* Update Snafu types and generated types
* `fork()` is an unsafe function
* Use `chrono` crate instead of `time` crate. As getting the system time in a multi-threaded application became significantly more annoying with the commits [here][1] and [here][2]. Whereas `chrono` does not need these configuration parameters.  And easy to use a single dependency everywhere where time formatting is needed.
* Update `clap` to new 4.0 APIs
* Various clippy warnings
* Fix regression in recent .len() > 0 refactorings, boolean flipped in:
  * `grammartec/src/tree.rs`
* Unpin nightly from a specific daily version of nightly. Most Rust developers keep a floating nightly install. Requiring an exact toolchain per project clutters installed toolchains and not trivial to chase down what project need what exact version.
* Force Address Sanitizer to not symbolize during fuzzing campaigns. Symbolizing is only useful to a human upon triaging ASAN crashes asynchronously of the fuzzing loop

[1]: https://github.com/time-rs/time/commit/77f8bfa8b0c37d8686b2ed2e81d93d028a3415e0
[2]: https://github.com/time-rs/time/commit/ec4057d6b8606ec7485380328cf3baedc9b60e54

### CODE REVIEW

**Comment by andreafioraldi** [comment]
@eqv wdyt?

**Comment by andreafioraldi** [comment]
I'd merge it if you agree

File: rust-toolchain [hunk 1]
```diff
@@ -1 +1 @@
-nightly-2022-10-17
```
Comment (domenukk) [inline rust-toolchain:1]: Isn't specifying a working nightly version safer? Things can break between versions

**Review by domenukk — APPROVED** [review]

File: rust-toolchain (hunk 1 above)
Comment (rchildre3) [reply rust-toolchain:1]: On the nightly yes, breakages could happen, but upon further inspection, the only reason to use nightly is only used for a test in `grammatec`. I will push up a modification to this branch to move to stable. All targets are building and all tests passing

**Comment by rchildre3** [comment]
Ready for re-review

**Pull request merged**